pkg/crlprovider/provider.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crlprovider
+
+import (
+	corecrl "github.com/notaryproject/notation-core-go/revocation/crl"
+)
+
+type CRLFetcherOptions struct {
+	// disableCache provide an configuration for user to disable the cache
+	// disableCache bool
+}
+
+type CRLCacheOptions struct {
+	// the default cacheRoot value is dir.PathCRLCache
+}
+
+// CRLProvider defines the interface for managing CRL operations.
+type CRLProvider interface {
```

Comment:
nit: CRLProvider seems not suitable here. Provider gives me the sense that it provides CRL.

Reply:
About to rename it to crlfactory